azurelinuxagent/common/protocol/extensions_goal_state_from_extensions_config.py

```diff
@@ -521,7 +521,7 @@ def __parse_and_add_extension_settings(settings_node, name, ext_handler, depends
             # There is no "extension name" for single Handler Settings. Use HandlerName for those
             ext.name = name
             ext.state = state
-            ext.sequenceNumber = seq_no
+            ext.sequenceNumber = int(seq_no)
```

Comment:
the current code represents the sequence number as a string. I decided to change that to an int, which seems more natural. Unit tests and End-to-End automation reported no errors, so this change seems safe. If you have concerns about it, please post them here and I can make the new code parse this as a string instead (if it makes sense)